File: Task 1/C++/Components/Product/Physical.cpp
#include "Physical.h"

Physical::Physical() : Games(){
    number_of_players = 0;
};      

Physical::Physical(std::string name, float cost, std::string supplier, float discount_amount, int packing_units, float cost_per_unit, std::string genre, int number_of_Players) 
        : Games(name, cost, supplier, discount_amount, packing_units, cost_per_unit, genre){
    this->number_of_players = number_of_Players;
}

Physical::Physical(const Physical& other) : Games(other) {
    this->number_of_players = other.number_of_players;
}

std::string Physical::display_product_info(){
    return Games::display_product_info() +
            "\nNumber of Players: " +std::to_string(number_of_players); 
}

File: Task 1/Java/logistics/src/main/java/com/logistics/Components/Product/Physical.java
package com.logistics.Components.Product;

public class Physical extends Games {

    private int numberOfPlayers;

    public Physical() {
        super();
        numberOfPlayers = 0;
    }

    public Physical(String name, float cost, String supplier, float discountAmount, int packingUnits, float costPerUnit, String genre, int numberOfPlayers) {
        super(name, cost, supplier, discountAmount, packingUnits, costPerUnit, genre);
        this.numberOfPlayers = numberOfPlayers;
    }

    // Copy Constructor
    public Physical(Physical other) {
        super(other);
        this.numberOfPlayers = other.numberOfPlayers;
    }

    // Clone method
    @Override
    public Physical clone() {
        return new Physical(this);
    }

    // Getters
    public int getNumberOfPlayers() { return numberOfPlayers; }

    // Setters
    public void setNumberOfPlayers(int numberOfPlayers) { this.numberOfPlayers = numberOfPlayers; }

    @Override
    public String display_product_info() {
        return super.display_product_info() + 
                "\nNumber of Players: " + numberOfPlayers;
    }
}
